docker/el7/Dockerfile

```diff
@@ -2,17 +2,45 @@ FROM centos:7
 
 RUN yum -y update && yum -y upgrade && yum -y install epel-release
 RUN yum -y install centos-release-scl
+RUN yum -y install devtoolset-9
 
 # Install build dependenices
-RUN yum -y install git devtoolset-7 llvm-toolset-7 llvm-toolset-7-cmake gtk-doc libtool lcov wget patchelf lsof
+RUN yum -y install git gtk-doc libtool lcov wget patchelf lsof
+RUN yum -y --enablerepo=extras install epel-release
+RUN yum -y install cmake3 ninja-build libstdc+
+RUN ln -s cmake3 /usr/bin/cmake
 
-# Fetch llvm and build libc++
-RUN cd /tmp/ && git clone https://github.com/llvm/llvm-project.git --branch release/9.x --single-branch --depth 1 \
-    && cd /tmp/llvm-project/ && mkdir build && cd build \
-    && source scl_source enable llvm-toolset-7 \
-    && cmake -G "Unix Makefiles" -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_PROJECTS='libcxx;libcxxabi' ../llvm/ \
-    && make -j5 cxx \
-    && make install-cxx install-cxxabi
+RUN cd /tmp \
+    && wget -O clang+llvm-8.0.1-x86_64-linux-gnu-ubuntu-14.04.tar.xz https://github.com/llvm/llvm-project/releases/download/llvmorg-8.0.1/clang+llvm-8.0.1-x86_64-linux-gnu-ubuntu-14.04.tar.xz \
+    && tar xvf clang+llvm-8.0.1-x86_64-linux-gnu-ubuntu-14.04.tar.xz \
+    && cd clang+llvm-8.0.1-x86_64-linux-gnu-ubuntu-* \
+    && cp -rn * /usr/local \
+    && cd /tmp && rm -rf clang+llvm-8.0.1-x86_64-linux-gnu-ubuntu-* \
+    && ln -s /usr/local/bin/lld /usr/local/bin/ld
+
+# Build libcxx to /tmp/libcxx-8.0.1.src/build/lib/libc++.so.1.0 with GLIBC-2.17
+RUN cd /tmp \
+    && wget -O libcxx-8.0.1.src.tar.xz https://github.com/llvm/llvm-project/releases/download/llvmorg-8.0.1/libcxx-8.0.1.src.tar.xz \
+    && tar xvf libcxx-8.0.1.src.tar.xz \
+    && cd libcxx-8.0.1.src \ 
+    && source scl_source enable devtoolset-9 \
+    && mkdir build \
+    && cd build \
+    && cmake ../ -DCMAKE_BUILD_TYPE=release -G "Unix Makefiles" . \
+    && make -j$(getconf _NPROCESSORS_ONLN)
```

Comment:
This will let the libraries in `tmp` directory which I think is not a good practice. Can you copy/install them in a permanent directory? If someone wants to use this image and robots the container I think it is not ensured the files will remains there